### ISSUE TITLE
[Welcome] Add new whisper type: Fallback

### ISSUE THREAD

**Issue by brandons209** [tmercswims/tmerc-cogs#99]
### I am submitting a...
```
 [ ] Bug report
 [X] Feature request or improvement for an existing cog
```

### Name of the cog in question:
Welcome

### Description of my issue/request:
I've added a new welcome join whisper type: Fallback. It tries to send the whisper message to the user in a DM, if it fails (for whatever reason, normally if the user has DM from server members turned off), it will send the whisper message to the channel set for Welcomes.

This is good so that those with that setting turned off they can still get the welcome message, and not miss out on important info.

Relevant code:

enums.py:
```python
class WhisperType(Enum):
  OFF = 'off'
  ONLY = 'only'
  BOTH = 'both'
  FALLBACK = 'fall'
```

welcome.py: 
Exception to throw if whisper fails
```python
class WhisperError(Exception):
    pass

class WelcomeCustom(getattr(commands, "Cog", object)):
  """Announce when users join or leave a server."""
```
```python
@commands.Cog.listener()
  async def on_member_join(self, member: discord.Member):
    """Listens for member joins."""

    guild = member.guild
    guild_settings = self.config.guild(guild)

    if await guild_settings.enabled() and await guild_settings.join.enabled():
      # join notice should be sent
      message_format = None
      if member.bot:
        # bot
        message_format = await guild_settings.join.bot()

      else:
        # only increment when it isn't a bot
        await self.__increment_count(guild, 'join')

        whisper_type = await guild_settings.join.whisper.state()
        if whisper_type != 'off':
          try:
            await self.__dm_user(member)
          except:
            if whisper_type == 'fall':
                message_format = await self.config.guild(member.guild).join.whisper.message()
                await self.__handle_event(guild, member, 'join', message_format=message_format)
                return

          if whisper_type == 'only' or whisper_type == 'fall':
            # we're done here
            return

      await self.__handle_event(guild, member, 'join', message_format=message_format)
```

```python
  async def __dm_user(self, member: discord.Member):
    """Sends a DM to the user with a filled-in message_format."""

    message_format = await self.config.guild(member.guild).join.whisper.message()

    try:
      await member.send(message_format.format(member=member, server=member.guild))
    except discord.Forbidden:
      log.error(
        ("Failed to send DM to member ID {0.id} (server ID {1.id}): insufficient permissions"
         "").format(member, member.guild)
      )
      raise WhisperError("Error.")
    except:
      log.error(
        ("Failed to send DM to member ID {0.id} (server ID {1.id})"
         "").format(member, member.guild)
      )
      raise WhisperError("Error.")
```

```python
@welcomeset_join_whisper.command(name='type')
  async def welcomeset_join_whisper_type(self, ctx: commands.Context, choice: WhisperType):
    """Set if a DM is sent to the new member.

    Options:
      off - no DM is sent
      only - only send a DM to the member, do not send a message to the channel
      both - send a DM to the member and a message to the channel
      fall - send a DM to the member, if it fails send the whisper message to the channel instead
    """

    guild = ctx.guild
    whisper_type = choice.value
    channel = await self.__get_channel(ctx.guild)

    await self.config.guild(guild).join.whisper.state.set(whisper_type)

    if choice == WhisperType.OFF:
      await ctx.send(
        ("I will no longer DM new members, and will send a notice to {0.mention}."
         "").format(channel)
      )
    elif choice == WhisperType.ONLY:
      await ctx.send(
        ("I will now only DM new members, and will not send a notice to {0.mention}."
         "").format(channel)
      )
    elif choice == WhisperType.BOTH:
      await ctx.send(
        ("I will now send a DM to new members, as well as send a notice to {0.mention}."
         "").format(channel)
      )
    elif choice == WhisperType.FALLBACK:
      await ctx.send(
        ("I will now send a DM to new members, and if that fails I will send the message to {0.mention}."
         "").format(channel)
      )
```

**Comment by tmercswims** [comment]
Hi! Thanks for the contribution, I like this idea a lot. Any chance you could open a [pull request](https://github.com/tmercswims/tmerc-cogs/pulls) with these changes? It will make it much easier for me to see the diff and better understand the changes that you made. Thanks again!

**Comment by brandons209** [comment]
Yes I can. I didn't even think to do that, exhaustion is setting in.

**Comment by tmercswims** [comment]
Closed by #101.
